ios-app/UI/TestReportViewController.swift

```diff
@@ -251,6 +251,10 @@ class TestReportViewController: UIViewController {
             presentingViewController?.presentingViewController as? ContentDetailPageViewController {
             
             goToContentDetailPageViewController(contentDetailPageViewController)
+        } else if let customTestViewController =
+                    self.presentingViewController as? CustomTestGenerationViewController {
+            
+            customTestViewController.presentingViewController?.dismiss(animated: true, completion: nil)
```

Comment:
This part is not clear, what are you doing? if possible extract it into function with the intention revealing name.